test/unit/org/apache/cassandra/cql3/PreparedStatementsTest.java

```diff
@@ -581,4 +653,126 @@ private void testPrepareWithBatchLWT(ProtocolVersion version) throws Throwable
+        ));
+        // populate every row
+        int numPartitions = 5;
+        int[][] model = new int[numPartitions][];
+        for (int writePk = 0; writePk < numPartitions; writePk++)
+        {
+            model[writePk] = new int[] {0, 0};
+            assertRowsNet(version, session.execute(writeOnly.bind(writePk, 0, 0)));
+        }
+
+        for (int writePk = 0; writePk < numPartitions; writePk++)
+        {
+            for (int readPk = 0; readPk < numPartitions; readPk++)
+            {
+                int[] expected = model[readPk];
+                int[] mutated = model[writePk];
+                for (boolean select : Arrays.asList(true, false))
+                {
+                    for (int retries = 0; retries < maxAttempts; retries++)
+                    {
+                        try
+                        {
+                            ResultSet rs = session.execute(select ? returnSelect.bind(readPk, writePk)
+                                                                  : returnRef.bind(readPk, writePk));
+                            assertRowsNet(version, rs, row(readPk, expected[0], expected[1]));
+                            break;
+                        }
+                        catch (WriteTimeoutException e)
+                        {
+                            logger.warn("Write timeout seen", e);
+                            if (retries >= maxAttempts - 1) throw e;
+                            Uninterruptibles.sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
+                        }
+                        finally
+                        {
+                            // update to account for counter bumps
+                            mutated[0]++;
+                            mutated[1] = mutated[1] + 2;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private static String txn(String... stmts)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("BEGIN TRANSACTION\n");
+        for (String stmt : stmts)
+        {
+            sb.append("  ").append(stmt);
+            if (!stmt.endsWith(";")) sb.append(';');
+            sb.append('\n');
+        }
+        sb.append("COMMIT TRANSACTION");
+        return sb.toString();
+    }
+
+    private static String batch(String... stmts)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("BEGIN BATCH\n");
+        for (String stmt : stmts)
+        {
+            sb.append("  ").append(stmt);
+            if (!stmt.endsWith(";")) sb.append(';');
+            sb.append('\n');
+        }
+        sb.append("APPLY BATCH");
+        return sb.toString();
+    }
+
+    private static List<String> columnNames(ResultSet rs)
+    {
+        return rs.getColumnDefinitions().asList().stream().map(d -> d.getName()).collect(Collectors.toList());
+    }
+
+    private static void updateTxnState()
```

Comment:
nit: I guess we should throw in a TODO to remove this once transactional metadata is in place? (The cache disabling bit could just be in a `@Before` method...I think.)

Reply:
adding `//TODO Remove this method once CEP-21 and CEP-15 integrate`

> (The cache disabling bit could just be in a @Before method...I think.)

the current pattern is to do this w/e we do a schema change, so figured best to be consistent...